src/com.docker.slirp/src/main.ml

```diff
@@ -45,10 +45,20 @@ let start_port_forwarding port_control_path vsock_path =
 
 module Slirp_stack = Slirp.Make(Vmnet.Make(Hostnet.Conn_lwt_unix))(Resolv_conf)
 
-let main_t socket_path port_control_path vsock_path db_path debug =
+let set_nofile nofile =
+  let open Sys_resource.Resource in
+  let soft = Limit.Limit nofile in
+  let (_, hard) = Sys_resource_unix.getrlimit NOFILE in
+  Log.info (fun f -> f "Setting soft fd limit to %d" nofile);
+  try Sys_resource_unix.setrlimit NOFILE ~soft ~hard with
+  | Errno.Error ex -> Log.warn (fun f -> f "setrlimit failed: %s" (Errno.string_of_error ex))
+  | _ -> ()
```

Comment:
Bit of a brain fart that, in that I was trying to express the concept of "otherwise continue" in what is clear a nonsense fashion. I'll drop it.